certify-service/src/main/java/io/mosip/certify/config/AppConfig.java

```diff
@@ -106,16 +111,18 @@ public void run(ApplicationArguments args) throws Exception {
         partnerMasterKeyRequest.setReferenceId(org.apache.commons.lang3.StringUtils.EMPTY);
         keymanagerService.generateMasterKey(objectType, partnerMasterKeyRequest);
         // Generate an Ed25519Key:
-        // 1. Generate a master key first to enable Keymanager to store the key.
-        KeyPairGenerateRequestDto storeKey = new KeyPairGenerateRequestDto();
-        storeKey.setApplicationId(Constants.CERTIFY_VC_SIGN_ED25519);
-        storeKey.setReferenceId(Constants.EMPTY_REF_ID);
-        keymanagerService.generateMasterKey("certificate", storeKey);
-        // 2. Generate an Ed25519 key later
-        KeyPairGenerateRequestDto ed25519Req = new KeyPairGenerateRequestDto();
-        ed25519Req.setApplicationId(Constants.CERTIFY_VC_SIGN_ED25519);
-        ed25519Req.setReferenceId(Constants.ED25519_REF_ID);
-        keymanagerService.generateECSignKey("certificate", ed25519Req);
+        if(pluginMode.equals("DataProvider")) {
+            // 1. Generate a master key first to enable Keymanager to store the key.
+            KeyPairGenerateRequestDto storeKey = new KeyPairGenerateRequestDto();
```

Comment:
Can we add logs like above and then say the CERTIFY_VC_SIGN_RSA KEY CHECK with equals and the same for CERTIFY_VC_SIGN_ED25519 KEY CHECK etc..

Reply:
Added above logs